src/crimson/os/seastore/seastore_types.h

```diff
@@ -0,0 +1,189 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <limits>
+#include <iostream>
+
+#include "include/denc.h"
+#include "include/buffer.h"
+#include "include/cmp.h"
+
+namespace crimson::os::seastore {
+
+using checksum_t = uint32_t;
+
+// Identifies segment location on disk, see SegmentManager,
+// may be negative for relative offsets
+using segment_id_t = uint32_t;
+constexpr segment_id_t NULL_SEG_ID =
+  std::numeric_limits<segment_id_t>::max() - 1;
```

Comment:
seems like an index, and not an ID, right? if so - maybe '..._idx'?

Reply:
It's not necessarily an index into any linear structure.  Moreover, I predict that we'll use non-contiguous subranges to represent segments on different devices.  Id would seem to carry less baggage.